docs/section19/19-24.md

```diff
@@ -0,0 +1,100 @@
+6. Where the largest exhaust source is less than 75% of the design outdoor airflow. This exception shall only be used if exhaust air energy recovery is not used in the proposed design. 
+7. Systems requiring dehumidification that employ energy recovery in series with the cooling coil. This exception shall only be used if exhaust air energy recovery and series-style energy recovery coils are not used in the proposed design. 
+
+**Rule Assertion:** Options are Pass/Fail/UNDETERMINED     
+**Appendix G Section:** G3.1.2.10 and exceptions 1-7        
+**90.1 Section Reference:** None  
+
+**Data Lookup:** None  
+
+**Evaluation Context:** Each HeatingVentilatingAirConditioningSystem  
+
+**Applicability Checks:** None   
+
+**Function Calls:**  
+1. get_dict_of_zones_and_terminal_units_served_by_hvac_sys()   
+2. get_list_hvac_systems_associated_with_zone()   
+3. Get_largest_exhaust_source_cfm()
+4. get_baseline_system_types()
+
+## Rule Logic:      
+- Create ditionary of baseline system types: `baseline_system_types_dict = get_baseline_system_types(B_RMI)`  
+- Get climate zone: `climate_zone = ASHRAE229.weather.climate_zone` 
+- Check if climate zone is (0-3) 0A, 0B, 1A, 1B, 2A, 2B, 3A, 3B, 3C, if so set ER_not_req_for_heating_sys equal to true: `if climate_zone in ["CZ0A", "CZ0B", "CZ1A", "CZ1B", "CZ2A", "CZ2B", "CZ3A", "CZ3B", "CZ3C"]: ER_not_req_for_heating_sys = True`             
+- Create dictionary of hvac systems, zones, and terminal units: `dict_of_zones_and_terminal_units_served_by_hvac_sys = get_dict_of_zones_and_terminal_units_served_by_hvac_sys(B_RMI)`  
+This section checks whether energy recovery was modeled, what the modeled enthalpy recovery ratio was, and gathers the information to assess whether it was supposed to be modeled (without exceptions considered)
+- For each hvac systen in the B_RMI: `for hvac in B_RMI:`  
+    - Reset ER_modeled boolean variable to false: `ER_modeld = False`  
+    - Create a fan system object: `fan_system_b = hvac.fan_system`  
+    - Get list of supply fan fan objects: `fan_sys_supply_list = list(fan_system_p.supply_fans)`  
+    - Cycle through the list of supply fans: `for fan_b in fan_sys_supply_list:`  
+        - Get the supply cfm and add to cumulative supply cfm as the logic cycles through the supply fans: `supply_cfm_b = supply_cfm_b + fan_b.design_airflow`  
+    - Get the minimum OA cfm (use min or max OA cfm?): `OA_cfm = fan_system_b.minimum_outdoor_airflow`  
+    - Calculate the OA fraction: `OA_fraction = OA_cfm/supply_cfm_b`  
+    - Check if there are any energy recovery objects associated with the fan system object and that the type is not equal to None: `if fan_sys_b.air_energy_recovery != Null and fan_sys_b.air_energy_recovery.energy_recovery_type != "NONE":` 
+        - Set energy recovery modeled boolean variable to true: `ER_modeled = True`  
+        - Get the enthalpy recovery ratio: `enthalpy_rec_ratio = fan_sys_b.air_energy_recovery.enthalpy_recovery_ratio`  
```

Comment:
To verify the 50% enthalpy recovery ratio we also need to check the percentage of total exhaust and supply go through the ERV device. I think we need to add AirEnergyRecovery.outdoor_air_flow_rate and AirEnergyRecovery.exhaust_flow_rate

Reply:
exhaust flow = OA flow then enthalpy ratio at the input value

Reply:
@claperle Create an issue to add these to the schema.

Reply:
I am still not sure how we would verify it without enthalpy data for the air streams at rated conditions. I would like to add these for the sake of 19-19 calculations but I am not sure we can verify this here.

Reply:
If we have these additional fields, we could calculate the recovery effectiveness as AirEnergyRecovery.enthalpy_recovery_ratio*AirEnergyRecovery.exhaust_flow_rate/(Total central and zonal exhaust flow for all zones served by the system)* AirEnergyRecovery.outdoor_air_flow_rate/(Total OA flow rates supplies to all zones served by this syste).

Reply:
This is still in progress.

Reply:
we decided that checking the enthalpy recovery ratio would be undetermined. This means that none of the outcomes can be pass for this rule.

How about we split this into two rules. One that just checks that energy recovery was modeled at all if required and another that checks that the enthalpy recovery ratio was modeled per the 50%. This way we can separate checking the enthalpy ratio part of G3.1.2.10. We would get some passes with the first rule but the second rule would always be undetermined and would just show all the hvac systems where energy recovery was modeled in the baseline where the enthalpy recovery ratio needs to be checked.

Reply:
Group on board with this. CML to implement.